### DISCUSSION TITLE
Search filters and searches for longevity

### DISCUSSION THREAD

**Discussion by cuathro** [standardnotes/forum#2689] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?**
Yes

**Is your feature request related to a problem? Please describe.**
Having many notes or complex notes (with a lot of text), the search on Linux and much worse on Mobile (Android) is totally useless without filters to make the search function useful.

**Describe the solution you'd like**
There are options for sorting, but not for filtering the search. 

Options such as:
- Search on note titles.
- Search in the contents of the note
- Search case-sensitive search
- Search by separate words or as exact phrase 

Options to control the results, i.e. arrows to go to the next result found in the note or to go back to the previous one.

**Describe alternatives you've considered**
Focusing search efforts and filters.

**Additional context**
Managing notes with longevity in mind should include these search features. Notes, with longevity in mind, tend to complexify their content, so do not ask for complex features, but for basic management features.

**Comment by JaspalSuri** [comment]
Hi @khlinuxdeveloper, while more search filtering options aren't a bad idea, you can assign tags to notes to categorize them, select a tag to narrow down the notes list, and then search for the note that you're looking for (either manually or by using the search bar).

**Comment by cuathro** [comment]
Hi @JaspalSuri,

I don't think that the solution is to create tags as an axis to reduce results, because the only thing that would be achieved is to lose the sense of tags (which is to classify in a -macro- way). Something essential and key for the notes (which are designed for longevity) is that they are usable and accessible, without a protocol with which you run out of life looking for what you took note of (being a thought, a text, a letter, a project, etc.). The search must have the results -micros-, and these are achieved with basic filters. It does not ask for filters of approximation (~), of adding (+) or removing results (-), indexing of notes (which would be ideal to have them), but now, the only thing that is requested are the basic and necessary filters.

I attach a document in which I explain graphically the problem and the results of the comparison between the current and the basic filters. I hope you can take it into account as a priority function. 

Best regards

PDF: [1433-StandardNotes-Filters.pdf](https://github.com/standardnotes/forum/files/7199486/1433-StandardNotes-Filters.pdf)

**Comment by Paul-5** [comment]
Hi @khlinuxdeveloper, I agree with you entirely. Search is fundamental functionality to a notes application such as this. If you can't search through the information you're entering then there's no point in entering it.

Having to label every contact and idea you create in your note to circumvent the lack of usable search functionality is not a usable solution.

**Comment by atmoio** [comment]
You can create custom filters/queries: https://standardnotes.com/help/42/how-do-i-view-a-list-of-untagged-notes-and-create-other-dynamic-filters

At some point we'll incorporate this into search directly so you won't need Folders.

**Comment by cuathro** [comment]
When I get to that point where this feature is implemented, I will find the secret ingredient of a cake within my 1500 notes on cooking recipes, so I can give @moughxyz a piece and share it with @Paul-5 hahaha. 

Regards.

**Comment by jrcacd** [comment]
Essential functionality.
Hope it doesn't take too long to implement.
